core/src/main/scala-2.13+/cats/instances/arraySeq.scala

```diff
@@ -103,12 +103,11 @@ private[cats] object ArraySeqInstances {
       override def foldMap[A, B](fa: ArraySeq[A])(f: A => B)(implicit B: Monoid[B]): B =
         B.combineAll(fa.iterator.map(f))
 
-      def traverse[G[_], A, B](fa: ArraySeq[A])(f: A => G[B])(implicit G: Applicative[G]): G[ArraySeq[B]] = {
-        def loop(i: Int): Eval[G[ArraySeq[B]]] =
-          if (i < fa.length) G.map2Eval(f(fa(i)), Eval.defer(loop(i + 1)))(_ +: _)
-          else Eval.now(G.pure(ArraySeq.untagged.empty))
-        loop(0).value
-      }
+      def traverse[G[_], A, B](fa: ArraySeq[A])(f: A => G[B])(implicit G: Applicative[G]): G[ArraySeq[B]] =
+        G.map(Chain.traverseViaChain(fa)(f))(_.iterator.to(ArraySeq.untagged))
```

Comment:
make traverse stack safe on ArraySeq. We missed this before.

Reply:
This one is eluding me a bit... How come the previous implementation was not stack safe if each its call to `loop` is suspended in `Eval`? Just curious, because I even tried this snippet:
```scala
  val arr1 = ArraySeq.iterate(1, 100)(_ + 1)
  val arr2 = arr1.traverse { a =>
    println(s"\n>>> a = $a")
    new Exception().printStackTrace(Console.out)
    a.some
  }
```
And seems that stack depth is kept as O(1) for the entire `traverse` evaluation.
Or am I missing something?

Reply:
Does that work when you make 100 something like 100000?

The fn call you are printing is at constant depth but there is more to it, there is also the mapping function on the map2.

That said, we have hit many issues with traverse stack safety in the past and I think the approach used here was the same there (but maybe I am wrong).

Reply:
> Does that work when you make 100 something like 100000?

Yes, it does even for 1,000,000 :) 

> The fn call you are printing is at constant depth but there is more to it, there is also the mapping function on the map2.
> 
> That said, we have hit many issues with traverse stack safety in the past and I think the approach used here was the same there (but maybe I am wrong).

I mean I'm not against the new implementation (perhaps it's supposed to be more performant). Just curious, because I tried out and didn't manage to screw up the old one :)